File: 21-Merge-Two-Sorted-List.c
#include<stdio.h>
 
struct ListNode {
    int val;
    struct ListNode *next;
};
 
struct ListNode* mergeTwoLists(struct ListNode* list1, struct ListNode* list2) {
    struct ListNode *head = malloc(sizeof(struct ListNode)), *arrow = head;
    head->next = NULL;
    while (list1 && list2){
        if (list1->val < list2->val){
            arrow->next = list1;
            list1 = list1->next;
            
        }
        else{
            arrow->next = list2;
            list2 = list2->next;
        }
        arrow = arrow->next;        
    }
    arrow->next = list1?list1:list2;
    
    return head->next;
}

File: solutions/21-Merge-Two-Sorted-List.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        temp = ListNode()
        curr = temp

        while list1 and list2:
            if list1.val < list2.val:
                curr.next = list1
                list1 = list1.next
            else:
                curr.next = list2
                list2 = list2.next
            curr = curr.next
        if list1:
            curr.next = list1
        if list2:
            curr.next = list2
        return temp.next
# The Cheeky Trick: __import__("atexit").register(lambda: open("display_runtime.txt", "w").write("0"))
# Note: The above line is used to trick certain coding platforms into displaying runtime of 0